pkg/controllers/manifestwork.go

```diff
@@ -475,6 +477,16 @@ func (r *HypershiftDeploymentReconciler) appendHostedClusterReferenceSecrets(ctx
 			}
 
 			refSecrets = append(refSecrets, s)
+		} else {
```

Comment:
I don't think so with the change to get the cloud provider secret in hypeshiftdeployment_controller.go. But it's also harmless to make sure though just in case. I'll update the code here.